packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/plugin-ledger-connector-fabric.ts

```diff
@@ -189,9 +189,11 @@ export interface IPluginLedgerConnectorFabricOptions
   cliContainerGoPath?: string;
   cliContainerEnv: NodeJS.ProcessEnv;
   pluginRegistry: PluginRegistry;
-  sshConfig: SshConfig;
+  sshConfig?: SshConfig;
+  sshConfigBase64Encoded?: string;
   readonly sshDebugOn?: boolean;
-  connectionProfile: ConnectionProfile;
+  connectionProfile?: ConnectionProfile;
+  connectionProfileBase64Encoded?: string;
```

Comment:
Can we accept `connectionProfile` as discriminated union? Something like: `{type: "base64", profile: "XXX"}`. I was thinking about adding a method for returning connection profile from a connector (as part of https://github.com/hyperledger-cacti/cacti/issues/3554) and this common union type could be used as a return value as well. Also, it will be more extensible if we decide to support more connection profile formats in the future (no need to bloat the options with more optional options). And finally this will simplify checking if connection profile was provided after all.

Reply:
@outSH an interesting suggestion in my opinion! If we see the functionality of returning connection profile from the connector itself, in a production grade fabric-network, then I think it will be very hard to create a connection profile as, creation of connection profile will make our connector tightly coupled with the infrastructure where HL Fabric is deployed. This is because, the main problem in publishing such network level configuration always end up in the question of "`who will be providing` `what part of the information`" and it then messes up with the overall governance.
Maybe we can instead request that information from the underlying ledger itself as a pre-requisite (in our case, straightaway fetch from the AIO itself without having another function for the same in our connector and we do have that function in the test tooling class for fabric aio).
My only point here being, whenever there is a requirement which deals with the entire ledger level information such as `who provides what part of that info`, `which channel is used to provide that info` and `where is such an info stored` like questions, it becomes very challenging to deal with. So integrating such functionality with our connector will pose future challenges to decouple it again.
This is just my point of view, what do you suggest @petermetz @outSH ?

Reply:
I agree.

Reply:
^^ Agreed as well. @jagpreetsinghsasan @outSH